### ISSUE TITLE
Installation statistics banner considers file:// dependences to be "downloaded"

### ISSUE THREAD

**Issue by octogonz** [pnpm/pnpm#1103]
### Code to reproduce the issue:

Run `pnpm install` for a **package.json** with an entry like this:

```json
{
  "@rush-temp/package-deps-hash": "file:./projects/package-deps-hash.tgz",
}
```

Then run `pnpm install` a second time.

### Expected behavior:

The install statistics banner should look like this:

```
Resolving: total 1252, reused 1252, downloaded 0, done
```

### Actual behavior:

The install statistics banner shows this:

```
Resolving: total 1252, reused 1251, downloaded 1, done
```

This seems trivial, but we have 120 tarball dependencies like this in our Rush repo. People occasionally ask me why it keeps downloading new stuff when they haven't added anything new.  I have to explain that it's "downloading" it from their local filesystem, which sounds weird.

### Additional information:

 - `node -v` prints: 8.9.4
 - Windows, OS X, or Linux?: Windows

**Comment by pgonzal** [comment]
This is a cosmetic/polish issue.  If it's not a simple 5 minute fix, don't bother.  :-)

**Comment by ExE-Boss** [comment]
Well, the `file://` protocol can be used with remote hosts (an empty (`file://…`) or `null` (`file:…`) hostname means `localhost`).

**Comment by ExE-Boss** [comment]
Yes (just push it through a&nbsp;parser to&nbsp;determine if it’s local or not).

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

Verified on pnpm 12.5.1: a `file:` tarball dependency counts as `reused`, not `downloaded`, on a repeat install.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
